M2/Macaulay2/m2/reals.m2

```diff
@@ -106,10 +106,13 @@ promote(RawRingElement,RingElement) := (x,R) -> new R from x
 promote(Number,InexactNumber) := (x,RR) -> promote(x,default RR)
 promote(ZZ,RR') := 
 promote(QQ,RR') := 
+promote(RR',RR') :=
```

Comment:
A bit confused:
1. I think there's a better way to have promote behave like identity if the input ring and new ring are identical.
2. The commit that changes this file says
> fix isPromotable ring family
> This reverts commit https://github.com/Macaulay2/M2/commit/63d7fc73d45757ad1f46b63845af1d377574f934.

But that commit doesn't seem to have anything to do with this file.

Reply:
the revert part of the message is an accident. will fix.
the addition of `promote(XX',YY')` is an unfortunate side-effect of the messy way that inexact fields are implemented in Macaulay2 (which one day I'd love to rewrite from scratch, because it's such a mess).
They're not actual functions though because they never actually get called, really one should make them dummy functions.